docs: add documentation for Hosted MCP Server in Hedera AI Agent Kit (#529)
hashgraph/hedera-docs: hedera/open-source-solutions/ai-studio-on-hedera/hosted-mcp-server.mdx

Reviewer: @skurzyp-blockydevs I would actually move this up a level, instead of being under ai-agent-kit put it under ai-studio : This is a separate service from our SDK

</Note>

<Note>
**Network:** The hosted server is connected to **Hedera Testnet only**. For Mainnet integrations, run the [self-hosted MCP server](https://github.com/hashgraph/hedera-agent-kit-js/tree/main/examples/modelcontextprotocol).
</Note>

## Connection Details

The server uses the MCP Streamable HTTP transport.

- **Endpoint:** `<HOSTED_SERVER_URL>/mcp`
- **Transport:** Streamable HTTP
- **Network:** Hedera Testnet

> Replace `<HOSTED_SERVER_URL>` with the official Hashgraph Hosted MCP Server URL once provided.

## Context Initialization

When connecting to the server, provide your Hedera Account ID via an HTTP header. This initializes the session context so every tool call knows which account to build transactions for.

- **Header Name:** `x-hedera-account-id`
- **Value:** Your Hedera Testnet Account ID (e.g., `0.0.12345`)

On the initialize request, the server queries the Hedera Mirror Node to fetch the associated public key and configures the agent's scope for your account. If the account cannot be found on the Mirror Node, the initialize request returns `400` with an explanatory error.

<Note>

> Never send private keys to the hosted server. Only your account ID is required — the server constructs unsigned transactions, and signing happens entirely on your side.

Reviewer: ```suggestion
Never send private keys to the hosted server. Only your account ID is required — the server constructs unsigned transactions, and returns bytes for you to sign and execute.
```
